### ISSUE TITLE
EBADF, bad file descriptor

### ISSUE THREAD

**Issue by VectorHo** [shelljs/shelljs#180]
on window platform，use exec，eg：

``` coffeescript
options =
        async: true
        timeout: 1000*60*10
        maxBuffer: 1024*1024*10

exec 'git init', options, (code, output) ->
        console.log('Exit code:', code);
        console.log('Program output:', output);
        if code != 0
          reject("Error: Git init failed:#{output}")
        else
          resolve()
```

```
throw e：
Error: EBADF, bad file descriptor
  at Error (native)
  at Object.fs.writeSync (fs.js:559:20)
  at SyncWriteStream.write (fs.js:1768:6)
  at Socket.<anonymous> (c:\Users\comeontom\.atom\packages\atom-butterfly\node_modules\shelljs\src\exec.js:113:22)
  at Socket.emit (events.js:129:20)
  at readableAddChunk (_stream_readable.js:161:16)
  at Socket.Readable.push (_stream_readable.js:125:10)
  at Pipe.onread (net.js:522:20)
```

**Comment by ariporad** [comment]
@VectorHo: Hi! would you mind re-testing this with the latest version of shelljs.

**Comment by nfischer** [comment]
^ from the git repo is best, since we haven't done a release in quite a while

**Comment by nfischer** [comment]
@VectorHo is this still a concern? We've released ShellJS v0.7, which should have a lot of improvements.

**Comment by nfischer** [comment]
If this is still an issue, please reopen
